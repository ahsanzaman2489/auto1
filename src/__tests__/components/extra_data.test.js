import React from 'react';
import ExtraDataComponent from '../../components/extraData';
import {shallow} from 'enzyme';
import renderer from 'react-test-renderer';


const setup = (props = {}, state = null) => {
    const defaultProps = {
        item: {
            stockNumber: 10172,
            color: "red",
            mileage: {"number": 182124, "unit": "km"},
            fuelType: "Petrol",
        }
    };
    const setUpProps = {...defaultProps, ...props};
    const wrapper = shallow(<ExtraDataComponent {...setUpProps}/>);
    const tree = () => renderer.create(<ExtraDataComponent {...setUpProps}/>).toJSON();
    if (state) wrapper.setState(state);
    return {wrapper, props: setUpProps, tree};
};


describe("Extra data component", () => {
    it('should match with snapshot', () => {
        const {tree} = setup();
        expect(tree()).toMatchSnapshot();
    });
    it('should render no data text if no data', () => {
        const {wrapper, props} = setup();
        expect(wrapper.find('li').length).toBe(4);
        expect(wrapper.find('ul').childAt(0).text()).toBe(`Stock # ${props.item.stockNumber}`);
        expect(wrapper.find('ul').childAt(1).text()).toBe(`${props.item.mileage.number} ${props.item.mileage.unit}`);
        expect(wrapper.find('ul').childAt(2).text()).toBe(props.item.fuelType);
        expect(wrapper.find('ul').childAt(3).text()).toBe(props.item.color);
    });
});


