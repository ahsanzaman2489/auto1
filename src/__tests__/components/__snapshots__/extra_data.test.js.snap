// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Extra data component should match with snapshot 1`] = `
<div>
  <ul>
    <li>
      Stock # 
      10172
    </li>
    <li>
      182124
       
      km
    </li>
    <li>
      Petrol
    </li>
    <li>
      red
    </li>
  </ul>
</div>
`;
